prime-router/src/main/kotlin/fhirengine/engine/FHIRConverter.kt

```diff
@@ -32,39 +33,33 @@ class FHIRConverter(
     settings: SettingsProvider = this.settingsProviderSingleton,
     db: DatabaseAccess = this.databaseAccessSingleton,
     blob: BlobAccess = BlobAccess(),
-    queue: QueueAccess = QueueAccess,
+    queue: QueueAccess = QueueAccess
 ) : FHIREngine(metadata, settings, db, blob, queue) {
 
     /**
-     * [message] is the incoming HL7 message to be turned into FHIR and saved
+     * Accepts a [message] in either HL7 or FHIR format
+     * HL7 messages will be converted into FHIR
+     * FHIR messages will be decoded and saved
+     *
+     * [message] is the incoming message to be turned into FHIR and saved
      * [actionHistory] and [actionLogger] ensure all activities are logged.
      */
     override fun doWork(
         message: RawSubmission,
         actionLogger: ActionLogger,
-        actionHistory: ActionHistory,
+        actionHistory: ActionHistory
     ) {
-        logger.trace("Processing HL7 data for FHIR conversion.")
-        try {
-            // create the hl7 reader
-            val hl7Reader = HL7Reader(actionLogger)
-
-            // get the hl7 from the blob store
-            val hl7messages = hl7Reader.getMessages(message.downloadContent())
-
-            if (actionLogger.hasErrors()) {
-                throw java.lang.IllegalArgumentException(actionLogger.errors.joinToString("\n") { it.detail.message })
-            }
-
-            // use fhir transcoder to turn hl7 into FHIR
-            val fhirBundles = hl7messages.map {
-                HL7toFhirTranslator.getInstance().translate(it)
-            }
+        val format = Report.getFormatFromBlobURL(message.blobURL)
```

Comment:
I do not see a reason not to do this, but when we add the new upcoming functionality then we will need to access the DB and get the whole report info, so this code may be redundant then.

Reply:
This is related to the conversation [here](https://github.com/CDCgov/prime-reportstream/pull/7543#discussion_r1040111421), so this will probably be resolved with whatever comes out of that.